### ISSUE TITLE
Potential bug in `kernel_indefinite_stack_integral!`

### ISSUE THREAD

**Issue by charleskawczynski** [CliMA/ClimateMachine.jl#1382]
# Description

It seems there's a bug in `kernel_indefinite_stack_integral!`. From [this branch](https://github.com/CliMA/ClimateMachine.jl/tree/ck/EDMFIntegralBug), run 

```
julia --project
include("test/Atmos/EDMF/edmf_kernels.jl")
```

We hit [this error](https://github.com/CliMA/ClimateMachine.jl/blob/ck/EDMFIntegralBug/test/Atmos/EDMF/edmf_kernels.jl#L962), despite having [ensured positivity](https://github.com/CliMA/ClimateMachine.jl/blob/ck/EDMFIntegralBug/test/Atmos/EDMF/edmf_kernels.jl#L936).

## Additional context

Add any other reasons why this should be addressed.

# For CLIMA Developers

- [ ] This feature can be added (if it cannot be, explain why in a comment below -- lack of technical expertise, not relevant to the scope of this project, too ambitious)
- [ ] There is a timeline for when this feature can be implemented
- [ ] The feature has been (or will be) implemented (Please link the PR)

**Comment by mwarusz** [comment]
I don't think that this shows there is a bug. I believe that having positive nodal values doesn't guarantee that the indefinite integral will be positive everywhere since we are doing interpolation onto a sub-interval.

**Comment by charleskawczynski** [comment]
Good point @mwarusz, thanks for taking a look. Should we just wrap the result `x` in `max(0, x)`, or is there a more elegant/correct way to handle this?

My first thought was to try `N_poly=2` or `N_poly=1`. I'm still seeing negative values for `N_poly=2`, `N_poly=1` doesn't seem to be giving negative values, but idk how long we can use that (also, idk if our kernels are valid for `N_poly=1`).
